arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/predef.kt

```diff
@@ -25,13 +24,13 @@ internal fun <A> Iterable<A>.deleteFirst(f: (A) -> Boolean): Pair<A, List<A>>? {
   return go(this, emptyList())
 }
 
-internal fun <A> Iterable<A>.uncons(): Pair<A, List<A>>? =
+fun <A> Iterable<A>.uncons(): Pair<A, List<A>>? =
   firstOrNull()?.let { Pair(it, drop(1)) }
 
-internal fun Iterable<*>.isEmpty(): Boolean =
+fun Iterable<*>.isEmpty(): Boolean =
```

Comment:
Same here, not sure if it's a good idea to publish this extension, specially given the fact it gives no clue about it's doing a fold over the entire data set in case is not a Collection

Reply:
Just double-checked, and this could be moved to Stream as `internal`. I did however really miss an entire set of these methods in the Kotlin Std given that you can find them in Scala Std.

Reply:
I didn't see a big issue here, since these are also all things we find Scala Std, Guava, Apache Common, etc
IMO there is a huge set of these `Iterable` operators missing from Kotlin's Std. We could duplicate `size`, and move some of the other methods to Stream.

Reply:
Moved to streams as internal and duplicated `size`.